packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -511,6 +512,79 @@ export type Deployment = {
   versions?: DeploymentVersion[];
 };
 
+// If deploymentId is provided tries fetch only that one deployment and finishes early when found
```

Comment:
If deploymentId is provided it tries to fetch only that one deployment and returns early when found